lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+pub struct ChunkedGelfDecoderOptions {
+    /// The timeout in milliseconds for a message to be fully received. If the timeout is reached, the
+    /// decoder will drop all the received chunks of the uncomplete message and start over.
+    ///  The default value is 5 seconds.
+    #[serde(
+        default = "default_timeout_millis",
+        skip_serializing_if = "vector_core::serde::is_default"
+    )]
+    pub timeout_millis: u64,
+
+    /// The maximum number of pending uncomplete messages. If this limit is reached, the decoder will start
+    /// dropping chunks of new messages. This limit ensures the memory usage of the decoder's state is bounded.
+    /// The default value is 1000.
+    #[serde(
+        default = "default_pending_messages_limit",
+        skip_serializing_if = "vector_core::serde::is_default"
+    )]
+    pub pending_messages_limit: usize,
+}
+
+impl Default for ChunkedGelfDecoderOptions {
+    fn default() -> Self {
+        Self {
+            timeout_millis: default_timeout_millis(),
+            pending_messages_limit: default_pending_messages_limit(),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct MessageState {
+    total_chunks: u8,
+    chunks: [Bytes; GELF_MAX_TOTAL_CHUNKS as usize],
+    chunks_bitmap: u128,
+    timeout_task: JoinHandle<()>,
+}
+
+impl MessageState {
+    pub fn new(total_chunks: u8, timeout_task: JoinHandle<()>) -> Self {
+        Self {
+            total_chunks,
+            chunks: DEFAULT_CHUNKS,
+            chunks_bitmap: 0,
+            timeout_task,
+        }
+    }
+
+    pub fn is_chunk_present(&self, sequence_number: u8) -> bool {
+        let chunk_bitmap_id = 1 << sequence_number;
+        self.chunks_bitmap & chunk_bitmap_id != 0
+    }
+
+    pub fn add_chunk(&mut self, sequence_number: u8, chunk: Bytes) {
+        let chunk_bitmap_id = 1 << sequence_number;
+        self.chunks[sequence_number as usize] = chunk;
+        self.chunks_bitmap |= chunk_bitmap_id;
+    }
+
+    pub fn is_complete(&self) -> bool {
+        self.chunks_bitmap.count_ones() == self.total_chunks as u32
+    }
+
+    pub fn retrieve_message(&mut self) -> Option<Bytes> {
+        if self.is_complete() {
+            self.timeout_task.abort();
+            let chunks = &self.chunks[0..self.total_chunks as usize];
+            let mut message = BytesMut::new();
+            for chunk in chunks {
+                message.extend_from_slice(chunk);
+            }
+            Some(message.freeze())
+        } else {
+            None
+        }
+    }
+}
+
+/// A codec for handling GELF messages that may be chunked. The implementation is based on [Graylog's GELF documentation](https://go2docs.graylog.org/5-0/getting_in_log_data/gelf.html#GELFviaUDP)
+/// and [Graylog's go-gelf library](https://github.com/Graylog2/go-gelf/blob/v1/gelf/reader.go).
+#[derive(Debug, Clone)]
+pub struct ChunkedGelfDecoder {
+    // We have to use this decoder to read all the bytes from the buffer first and don't let tokio
```

Comment:
I don't know if this comment is well explained (or if it has to be removed), tell me what you think please.
That was a bug I faced while testing with big messages. 

we have to rely on the `decode_eof` method for the decoding and not in the `decode` method, such as the `ByteDecoder` does.

In order to now repeat code with that decoder, I used it as an inner decoder since we want the same behaviour; read all bytes from the current buffer.

Reply:
I think it makes sense. This framer needs to read the entire "packet". As you mentioned in another comment, it may still be prudent to have a configurable `max_length` here to avoid corrupt input ballooning memory usage.